### ISSUE TITLE
295953948 Gobbledegook error messages (Ali)

### ISSUE THREAD

**Issue by FreeREGcomputer** [FreeUKGen/FreeUKRegProductIssues#1527]
Issue reported by **alsibk** at 2024-05-18 09:25:48 UTC
Time: 2024-05-18T09:09:49+00:00
Session ID: 783174de3b0b46c5972d399985d48d19
Problem Page URL: [/freereg1_csv_files?locale=en](/freereg1_csv_files?locale=en)
Previous Page URL: [https://www.freereg.org.uk/csvfiles/new](https://www.freereg.org.uk/csvfiles/new)
Reported Issue:
Please---we volunteer CCs have enough problems without having to interpret such error messages.

Please could we have:

The line in question quoted in full in the error message.

A comprehensive list of examples of illegal quoting to which we may refer.

 "We were unable to complete the file BarbaraT OTHSHIBU4.csv. because Illegal quoting in line 727. Please contact your coordinator or the System Administrator with this message. ["/opt/rubies/2.6.10/lib/ruby/2.6.0/csv/parser.rb:879:in `parse_quotable_robust'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv/parser.rb:818:in `block in parse_quotable_loose'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv/parser.rb:31:in `block in each_line'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv/parser.rb:28:in `each_line'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv/parser.rb:28:in `each_line'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv/parser.rb:779:in `parse_quotable_loose'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv/parser.rb:303:in `parse'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv.rb:1236:in `each'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv.rb:1236:in `each'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv.rb:1245:in `to_a'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv.rb:1245:in `read'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/csv.rb:685:in `parse'", "/home/apache/hosts/freereg2/production/lib/new_freereg_csv_update_processor.rb:714:in `extract_the_array_of_lines'", "/home/apache/hosts/freereg2/production/lib/new_freereg_csv_update_processor.rb:895:in `slurp_the_csv_file'", "/home/apache/hosts/freereg2/production/lib/new_freereg_csv_update_processor.rb:345:in `a_single_csv_file_process'", "/home/apache/hosts/freereg2/production/lib/new_freereg_csv_update_processor.rb:111:in `block in activate_project'", "/home/apache/hosts/freereg2/production/lib/new_freereg_csv_update_processor.rb:109:in `each'", "/home/apache/hosts/freereg2/production/lib/new_freereg_csv_update_processor.rb:109:in `activate_project'", "/home/apache/hosts/freereg2/production/lib/new_freereg_csv_update_processor.rb:79:in `process_activate_project'", "/home/apache/hosts/freereg2/production/lib/tasks/build.rake:565:in `block (2 levels) in '", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/task.rb:279:in `block in execute'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/task.rb:279:in `each'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/task.rb:279:in `execute'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/airbrake-11.0.1/lib/airbrake/rake.rb:19:in `execute'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/task.rb:219:in `block in invoke_with_call_chain'", "/opt/rubies/2.6.10/lib/ruby/2.6.0/monitor.rb:235:in `mon_synchronize'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/task.rb:199:in `invoke_with_call_chain'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/task.rb:188:in `invoke'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:160:in `invoke_task'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:116:in `block (2 levels) in top_level'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:116:in `each'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:116:in `block in top_level'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:125:in `run_with_threads'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:110:in `top_level'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:83:in `block in run'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:186:in `standard_exception_handling'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/lib/rake/application.rb:80:in `run'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/gems/rake-13.0.1/exe/rake:27:in `'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/bin/rake:23:in `load'", "/home/apache/hosts/freereg2/production/vendor/bundle/ruby/2.6.0/bin/rake:23:in `
'"]
Rescued from crash Illegal quoting in line 727. "    

![Screenshot](https://www.freereg.org.uk)

**Comment by Vino-S** [comment]
REady for testing in test3

**Comment by DeniseColbert** [comment]
@Vino-S can you add @stsccfr to this repo so dale can be assigned to test?

**Comment by stsccfr** [comment]
I created a small test file that contained a field consisting of a double-quoted string followed by a single character outside the quotes, which is not allowed in CSV.  I uploaded the file to FreeREG to verify that it would reject the file for the expected reason, which is "Any value after quoted field isn't allowed".  The email I received included the long crash dump as expected.  I then uploaded the file to test3, and this time the email contained the basic error message but no crash dump.  So I think this issue has been resolved.
